### PR TITLE
feat/STADTPULS 589/records rls delete insert update

### PR DESCRIPTION
**Pull request by ff6347**
- chore(configs): Minimal config change for eslint
- feat(RLS): Adds functions and policies for records
- test(RLS): Add tests for update and insert records

## Todo

- [x] UPDATE using PATCH request
- [x] DELETE using DELETE request
- [x] INSERT using POST request

### CODE REVIEW

**Comment by vogelino** [comment]
@ff6347 could we already merge this for delete? We need the RLS policy on records to work in order for the feature to be merged. We could do the PATCH one in a separate branch. What do you think?

**Comment by ff6347** [comment]
> @ff6347 could we already merge this for delete? We need the RLS policy on records to work in order for the feature to be merged. We could do the PATCH one in a separate branch. What do you think?

Yes we can. But we are still applying migrations manually. So a merge wont update the production db yet

**Comment by vogelino** [comment]
> > @ff6347 could we already merge this for delete? We need the RLS policy on records to work in order for the feature to be merged. We could do the PATCH one in a separate branch. What do you think?
> 
> Yes we can. But we are still applying migrations manually. So a merge wont update the production db yet

Ok great. I think I would do it anyway because when we run github actions in the frontend repo, we setup the backend using this repo. When we then run the cypress tests, the RLS are those of the main branch, meaning that deleting records don't work.

File: supabase-docker-compose/dockerfiles/postgres/docker-entrypoint-initdb.d/40-triggers-and-rls.sql [hunk 1]
```diff
@@ -142,9 +142,9 @@ create policy "Allow individual insert access" on public.records for
 insert with check (public.owns_sensor (auth.uid (), "sensor_id"));
 -- update
 create policy "Allow individual update access" on public.records for
-update using (public.owns_record (auth.uid (), "sensor_id")) with CHECK (public.owns_record (auth.uid (), "sensor_id"));
+update using (public.owns_record (auth.uid (), "id")) with CHECK (public.owns_record (auth.uid (), "id"));
```
Comment (vogelino) [inline supabase-docker-compose/dockerfiles/postgres/docker-entrypoint-initdb.d/40-triggers-and-rls.sql:145]: OMG that was it!? Mamamia! Hehehe

**Review by vogelino — DISMISSED** [review]
Good catch! That was some adventure...

File: supabase-docker-compose/dockerfiles/postgres/docker-entrypoint-initdb.d/40-triggers-and-rls.sql (hunk 1 above)
Comment (ff6347) [reply supabase-docker-compose/dockerfiles/postgres/docker-entrypoint-initdb.d/40-triggers-and-rls.sql:145]: hahaha

**Pull request merged**